src/nvim/tui/tui.c

```diff
@@ -1448,7 +1448,27 @@ static void patch_terminfo_bugs(TUIData *data, const char *term,
     // teminfo entries.  See
     // https://github.com/gnachman/iTerm2/pull/92 for more.
     // xterm even has an extended version that has a vertical bar.
-    if (true_xterm    // per xterm ctlseqs doco (since version 282)
+    if (konsole) {
+      // Konsole uses an idiosyncratic escape code to set the cursor shape and
+      // does not support DECSCUSR.  This makes Konsole set up and apply a
+      // nonce profile, which has side-effects on temporary font resizing.
+      // In an ideal world, Konsole would just support DECSCUSR.
+      data->unibi_ext.set_cursor_style = (int)unibi_add_ext_str(ut, "Ss",
+          TMUX_WRAP(tmux, "\x1b]50;CursorShape=%?"
+          "%p1%{3}%<" "%t%{0}"    // block
+          "%e%p1%{5}%<" "%t%{2}"  // underline
+          "%e%{1}"                // everything else is bar
+          "%;%d;BlinkingCursorEnabled=%?"
+          "%p1%{1}%<" "%t%{1}"  // Fortunately if we exclude zero as special,
+          "%e%p1%{1}%&"  // in all other cases we can treat bit #0 as a flag.
+          "%;%d\x07"));
+      if (-1 == data->unibi_ext.reset_cursor_style) {
+          data->unibi_ext.reset_cursor_style = (int)unibi_add_ext_str(ut, "Se",
+                                                                      "");
+      }
+      unibi_set_ext_str(ut, (size_t)data->unibi_ext.reset_cursor_style,
+          "\x1b]50;\x07");
+    } else if (true_xterm    // per xterm ctlseqs doco (since version 282)
```

Comment:
Embedding decisions in the order of these conditions is fragile. Explicit is better. It also avoids churn.

Why wouldn't this work:

    if (!konsole && (true_xterm ...

Reply:
Ok, I've rewrite the patch with your advice. Seems to work too.